src/libraries/System.Text.RegularExpressions/tests/System.Text.RegularExpressions.Generators.Tests/RegexGeneratorParserTests.cs

```diff
@@ -556,6 +556,46 @@ partial class C
                 }", compile: true, additionalRefs: new[] { MetadataReference.CreateFromImage(referencedAssembly) }));
         }
 
+        [Fact]
+        public async Task WithGroups_ShouldDisplayGroupName()
+        {
+            string code = @"
+                using System.Text.RegularExpressions;
+                partial class C1
+                {
+                    [RegexGenerator(""^(?<LINE>[0-9]*)-(?<ENDLINE>[0-9]*)$"")]
+                    private static partial Regex A();
+                }
+            ";
+            LanguageVersion langVersion = LanguageVersion.Preview;
+            var proj = new AdhocWorkspace()
+               .AddSolution(SolutionInfo.Create(SolutionId.CreateNewId(), VersionStamp.Create()))
+               .AddProject("RegexGeneratorTest", "RegexGeneratorTest.dll", "C#")
+               .WithMetadataReferences(s_refs)
+               .WithCompilationOptions(new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary)
+               .WithNullableContextOptions(NullableContextOptions.Enable))
+               .WithParseOptions(new CSharpParseOptions(langVersion))
+               .AddDocument("RegexGenerator.g.cs", SourceText.From(code, Encoding.UTF8)).Project;
+
+            Assert.True(proj.Solution.Workspace.TryApplyChanges(proj.Solution));
+
+            Compilation? comp = await proj!.GetCompilationAsync(CancellationToken.None).ConfigureAwait(false);
+            Debug.Assert(comp is not null);
```

Comment:
It's unusual to use Debug.Assert in tests. We generally want them to fail in retail as well. Consider Assert.NotNull instead.